### ISSUE TITLE
Investigate and reduce garbage collector pressure (UObject pooling?)

### ISSUE THREAD

**Issue by nithinp7** [CesiumGS/cesium-unreal#980]
After #975, there will be significantly fewer and less frequent frame-rate drops due to Cesium for Unreal. One remaining source of frame-rate interruptions (while relatively infrequent) remains to be the garbage collector. There is often a huge (sometimes >1s hitch) while the GC is doing "reachability analysis", presumably traversing an enormous tree of UObject references, looking for objects to delete. After the reachability analysis concludes, there is a long period where each game thread tick invokes "incremental GC purging" - which is a time-allotted task that attempts to keep the frame-rate roughly at 30fps but the pressure invariably pushes the frame-rate down on average.

I'm quite confident that we are cleaning up all of our heavy resources (buffers, textures, GPU resource references, etc.). Tracing memory usage shows that texture and mesh memory naturally go up and down on their own due to our memory management, even _without_ GC. There is a separate "Untracked" category of memory usage that Unreal shows however which usually gets up to 2-3GB-ish, at which point the GC runs. The GC bookmark in the memory trace always immediately corresponds to a sharp decrease in the untracked category (in the below case it went down from 2.4GB to 1.6GB). 

##### The red "Untracked" category responds sharply to GC - the other categories are largely unaffected.
![GCMemoryTrace](https://user-images.githubusercontent.com/6706316/195223131-e3775e0c-c3d0-48a9-9593-a86158899b1e.PNG)

##### The timeline shows a 629ms hitch during the GC reachability analysis and a consistently lower frame-rate for a period after (incremental GC purge is running every frame at that point).
![GCHitch](https://user-images.githubusercontent.com/6706316/195224465-1dec71b5-e95a-486f-8875-655f3a27bb10.PNG)

I suspect that this untracked category may be an accumulation of abandoned UObjects that we create for gltf primitive components and textures that we eventually don't need. While the render resources and buffers are freed, we have so far had no clean way to "free" UObjects short of abandoning them for the garbage collector.

If the above suspicion is true (and it would have to be investigated to confirm it), then the most sensible solution seems to be to pool UObjects (gltf components, gltf primitive components, UTexture2D objects, and any other Cesium-created UObjects I'm forgetting about). If the untracked category corresponds to abandoned UObjects, this would likely entirely eliminate the GC pressure we are seeing - if investigation shows otherwise I'm sure someone will edit this or create a new issue!

**Comment by nithinp7** [comment]
Today I was profiling a scenario with a high-detailed photogrammetry tileset of a building (the max sse was lowered as well) and CWT with Bing. I was flying around the building, loading in tons of geometry then left the area entirely and just flew around the world. After the initial period before the memory usage reached a critical point the frame-rate was mostly fine. Once it hit a critical point, the GC ran causing a long hitch. Then - notably - I flew away from the high-detailed tileset and there continued to be smaller "Incremental GC Purge" hitches. These hitches did not stop for the rest of the scenario.

One issue here that has already been [previously identified](https://github.com/CesiumGS/cesium-native/issues/246), is that the high-detailed tileset has an in-memory cache that is continuing to take up space even once it is no longer needed (when we are flying around the rest of the world). The second discovery however, is that when overall memory usage is at a critical point, Unreal insists on doing garbage collection until it goes down (first with a long reachability analysis and then with a continual "incremental garbage collection"). When Unreal does this, it is not actually freeing any heavy resources since we take care of that on our own. So why is Unreal spending so much time on it? I suspect it's because there continue to be tons of unreferenced UObjects that Unreal mistakenly thinks _might_ be responsible - so it has to check and destroy all of them, only to find none of them free up significant memory.
